src/app/main/apps/vehicles/VehicleDialog.js

```diff
@@ -1,346 +1,315 @@
+              name="plate_number"
+              render={({ field }) => (
+                <TextField
+                  {...field}
+                  className="mb-24"
+                  label="Plate Number"
+                  id="plate_number"
+                  error={!!errors.plate_number}
+                  helperText={errors?.plate_number?.message}
+                  variant="outlined"
+                  required
+                  fullWidth
+                />
+              )}
+            />
+          </div>
 
-//           <div className="flex">
-//             <div className="min-w-48 pt-20">
-//               <Icon color="action">email</Icon>
-//             </div>
-//             <Controller
-//               control={control}
-//               name="email"
-//               render={({ field }) => (
-//                 <TextField {...field} className="mb-24" label="Email" id="email" variant="outlined" fullWidth />
-//               )}
-//             />
-//           </div>
+          <div className="flex">
+            <div className="min-w-48 pt-20">
+              <Icon color="action">settings</Icon>
+            </div>
+            <Controller
+              control={control}
+              name="engine_number"
+              render={({ field }) => (
+                <TextField
+                  {...field}
+                  className="mb-24"
+                  label="Engine Number"
+                  id="engine_number"
+                  error={!!errors.engine_number}
+                  helperText={errors?.engine_number?.message}
+                  variant="outlined"
+                  required
+                  fullWidth
+                />
+              )}
+            />
+          </div>
 
-//           <div className="flex">
-//             <div className="min-w-48 pt-20">
-//               <Icon color="action">domain</Icon>
-//             </div>
-//             <Controller
-//               control={control}
-//               name="company"
-//               render={({ field }) => (
-//                 <TextField {...field} className="mb-24" label="Company" id="company" variant="outlined" fullWidth />
-//               )}
-//             />
-//           </div>
+          <div className="flex">
+            <div className="min-w-48 pt-20">
+              <Icon color="action">local_gas_station</Icon>
+            </div>
+            <Controller
+              control={control}
+              name="fuel_type"
+              render={({ field }) => (
+                <TextField
+                  {...field}
+                  className="mb-24"
+                  label="Fuel Type"
+                  id="fuel_type"
+                  error={!!errors.fuel_type}
+                  helperText={errors?.fuel_type?.message}
+                  variant="outlined"
+                  required
+                  fullWidth
+                />
+              )}
+            />
+          </div>
+        </DialogContent>
```

Comment:
this fuel_type field should be Dropdown selection of 4 different options only. Otherwise it fails to save (500 error) when changing value and trying to save invalid data